src/platform/plugins/shared/discover/public/application/main/components/tabs_view/tabs_view.tsx

```diff
@@ -8,56 +8,68 @@
  */
 
 import { type TabItem, UnifiedTabs, TabStatus } from '@kbn/unified-tabs';
-import React, { useRef, useState } from 'react';
+import React, { useState } from 'react';
 import { pick } from 'lodash';
-import type { DiscoverSessionViewRef } from '../session_view';
+import { isOfAggregateQueryType } from '@kbn/es-query';
 import { DiscoverSessionView, type DiscoverSessionViewProps } from '../session_view';
 import {
   CurrentTabProvider,
   createTabItem,
   internalStateActions,
   selectAllTabs,
+  selectTabRuntimeState,
   useInternalStateDispatch,
   useInternalStateSelector,
 } from '../../state_management/redux';
 import { useDiscoverServices } from '../../../../hooks/use_discover_services';
+import { FetchStatus } from '../../../types';
 
-export const TabsView = ({
-  initialTabId,
-  sessionViewProps,
-}: {
-  initialTabId: string;
-  sessionViewProps: DiscoverSessionViewProps;
-}) => {
+export const TabsView = (props: DiscoverSessionViewProps) => {
   const services = useDiscoverServices();
   const dispatch = useInternalStateDispatch();
   const allTabs = useInternalStateSelector(selectAllTabs);
-  const [currentTabId, setCurrentTabId] = useState(initialTabId);
+  const currentTabId = useInternalStateSelector((state) => state.tabs.unsafeCurrentId);
   const [initialItems] = useState<TabItem[]>(() => allTabs.map((tab) => pick(tab, 'id', 'label')));
-  const sessionViewRef = useRef<DiscoverSessionViewRef>(null);
 
   return (
     <UnifiedTabs
       services={services}
       initialItems={initialItems}
-      onChanged={async (updateState) => {
-        await dispatch(
-          internalStateActions.updateTabs({
-            currentTabId,
-            updateState,
-            stopSyncing: sessionViewRef.current?.stopSyncing,
-          })
-        );
-        setCurrentTabId(updateState.selectedItem?.id ?? currentTabId);
-      }}
+      onChanged={(updateState) => dispatch(internalStateActions.updateTabs(updateState))}
       createItem={() => createTabItem(allTabs)}
```

Comment:
Noticed that duplicating a tab does not reapply its query/filters. We could look into it in a follow up work.

Reply:
True, I haven't implemented that portion yet. Created a dedicated issue for it here: #217582.